File: 스프링/김영한 스프링강의 - 2차/spring-security/src/main/java/dompoo/spring_security/member/MemberController.java
package dompoo.spring_security.member;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class MemberController {

    private final MemberService memberService;

    @PostMapping("/member")
    public Member saveMember(@RequestBody Member member) {
        return memberService.save(member);
    }

    @GetMapping("/member/{id}")
    public Member getMember(@PathVariable Long id) {
        return memberService.findById(id);
    }

    @GetMapping("/members")
    public List<Member> getMemberList() {
        return memberService.findAll();
    }
}

File: src/main/kotlin/dompoo/kopringdemo/api/MemberController.kt
package dompoo.kopringdemo.api

import dompoo.kopringdemo.api.response.MemberFindResponse
import dompoo.kopringdemo.api.response.MemberSaveResponse
import dompoo.kopringdemo.service.MemberService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
class MemberController(
	private val memberService: MemberService
) {
	@PostMapping("/member")
	fun saveMember(@RequestBody request: MemberSaveRequest): ResponseEntity<MemberSaveResponse> =
		ResponseEntity.ok().body(MemberSaveResponse.from(memberService.saveMember(request)))

	@GetMapping("/member/{id}")
	fun findMember(@PathVariable id: Long): ResponseEntity<MemberFindResponse> =
		ResponseEntity.ok().body(MemberFindResponse.from(memberService.findMemberById(id)))

}